sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -741,12 +741,16 @@ public void Dispose()
         /// </summary>
         ///
         /// <param name="expirationTimeUtc">The date/time, in UTC, that the current authorization is expected to expire.</param>
+        /// <param name="currentTimeUtc">The current date/time, in UTC.  If not specified, the system time will be used.</param>
         ///
         /// <returns>The interval after which authorization should be refreshed.</returns>
         ///
-        protected virtual TimeSpan CalculateLinkAuthorizationRefreshInterval(DateTime expirationTimeUtc)
+        protected virtual TimeSpan CalculateLinkAuthorizationRefreshInterval(DateTime expirationTimeUtc,
+                                                                             DateTime? currentTimeUtc = null)
         {
-            var refreshDueInterval = (expirationTimeUtc.Subtract(DateTime.UtcNow)).Add(AuthorizationRefreshBuffer);
+            currentTimeUtc ??= DateTime.UtcNow;
+
+            var refreshDueInterval = (expirationTimeUtc.Subtract(AuthorizationRefreshBuffer)).Subtract(currentTimeUtc.Value);
```

Comment:
The intent of this calculation is to use the buffer to adjust the expiration time such that we request a token earlier than the actual expiration so that tokens are renewed before they expire.

Reply:
Is it expected that if, for some reason, `refreshDueInterval` is less than `MinimumAuthorizationRefresh` that we prefer to choose the `MinimumAuthorizationRefresh` even if that means going past `expirationTimeUtc`?

Reply:
It is; this was a guard that we preserved from the T1 codebase to prevent spamming authorization requests.   In the case where authorization is denied, the client will handle the rejection and retry according to the policy; while it may cause throughput to slow down while recovery happens, it should be transparent to callers.

Pragmatically, the service should be responding with a reasonably long window.  IIRC, the current service implementation uses a 15 or 20 minute window.